cmd/lncli/commands.go

```diff
@@ -3160,3 +3162,149 @@ func forwardingHistory(ctx *cli.Context) error {
 	printRespJSON(resp)
 	return nil
 }
+
+var delegateMacaroonCommand = cli.Command{
+	Name:      "delegatemacaroon",
+	Category:  "Macaroons",
+	Usage:     "Delegates a macaroon by adding restrictions to an " +
+		"existing one",
+	ArgsUsage: "[format] [timeout] [ip_address]",
+	Description: `
+	Take an existing macaroon (use --macaroonpath) as a template, add
+	restrictions to it and then serialize it either in JSON (--format=json)
+	or binary (--format=binary) format.
+	`,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name: "format",
+			Usage: "the format to serialize the macaroon in, " +
+				"must be either 'json' or 'binary'",
+			Value: "binary",
```

Comment:
In testing this, it appears to default to `json` even though the default is set to binary.
```
$ lncli -n simnet delegatemacaroon --ip_address 127.0.0.2     │ing
{                                                                                         │2018-08-28 15:42:34.709 [ERR] DISC: unable to rebroadcast s
        "c": [                                                                            │tale channels: error while retrieving outgoing channels: no
                {                                                                         │ graph edges exist
                        "i": "ipaddr 127.0.0.2"                                           │2018-08-28 15:42:34.816 [INF] CRTR: Filtering chain using 0
                }                                                                         │ channels active
        ],                                                                                │2018-08-28 15:42:34.816 [INF] CRTR: Prune tip for Channel G
        "l": "lnd",                                                                       │raph: height=1500, hash=15b1caeddf87e4046661fe0a50a7445501f
        "i64": "AwoQQp_pVSCfaJXVA-zJtvQq2RIBMBoWCgdhZGRyZXNzEgRyZWFkEgV3cml0ZRoTCgRpbmZvEg│fb8a1e1bbab63afa17ae22525f70b                             
RyZWFkEgV3cml0ZRoXCghpbnZvaWNlcxIEcmVhZBIFd3JpdGUaFgoHbWVzc2FnZRIEcmVhZBIFd3JpdGUaFwoIb2Zm│2018-08-28 15:42:34.821 [INF] CMGR: Server listening on [::
Y2hhaW4SBHJlYWQSBXdyaXRlGhYKB29uY2hhaW4SBHJlYWQSBXdyaXRlGhQKBXBlZXJzEgRyZWFkEgV3cml0ZQ",  │]:9735
        "s64": "ugWq8Ib_KyCLATbzAffsqXMM3UKiJ1rxnIJDialeqhM"                              │2018-08-28 15:42:34.827 [INF] SRVR: Auto peer bootstrapping
}
```

Still working on testing this.